### PR TITLE
All uint tests

### PR DESCRIPTION
**Pull request by AlaaAlsalem**
- Follow gitFlow
- All unit tests pass
- All models have been validated 
- Use Rspec for testing

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    it { should have_many(:posts).with_foreign_key('author_id') }
+    it { should have_many(:comments).with_foreign_key('author_id') }
+    it { should have_many(:likes).with_foreign_key('author_id') }
+  end
+
+  describe 'validations' do
+    it { should validate_presence_of(:name) }
+    it { should validate_numericality_of(:posts_counter).only_integer.is_greater_than_or_equal_to(0) }
+  end
+end
```
Comment (codepantha) [inline spec/models/user_spec.rb]: - [ ] Please add a spec that tests the `three_latest_posts` method in your `User` model. It is a requirement to add tests for all of your models methods and validations as you can see below 👇 
image

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  describe 'private methods' do
+    let(:post) { create(:post) }
+
+    describe '#update_posts_counter' do
+      let(:author) { post.author }
+
+      it 'should increment posts_counter of the author' do
+        expect { post.send(:update_posts_counter) }.to change { author.reload.posts_counter }.by(1)
+      end
+    end
```
Comment (codepantha) [inline spec/models/post_spec.rb:40]: - [ ] Please test the `five_latest_comments` method in your `Post` model as it is a requirement that you test all custom methods in your models 👍

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @AlaaAlsalem 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Added model validations 📌 
✔️ Added unit tests 📌 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Please consider getting a detailed MIT license  [here](https://opensource.org/license/mit/). Currently, below 👇 is all you have in your license file
image


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello @AlaaAlsalem  👋 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request ⚠️ 
- [ ] Your Linter checks are failing on GitHub as seen in the image below. Please make sure all linter checks are passing before you ask fo a review :+1: 
image


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by DeliceLydia — APPROVED** [review]
## Hi @AlaaAlsalem, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 

## Highlights
- Addressed changes requested



### Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**